drivers/shared/src/vic_time.c

```diff
@@ -0,0 +1,794 @@
+/******************************************************************************
+ * @section DESCRIPTION
+ *
+ * VIC time and calendar module
+ *
+ * Ported from https://github.com/Unidata/netcdf4-python/blob/master/netcdftime/netcdftime.py
+ *
+ * @section LICENSE
+ *
+ * The Variable Infiltration Capacity (VIC) macroscale hydrological model
+ * Copyright (C) 2014 The Land Surface Hydrology Group, Department of Civil
+ * and Environmental Engineering, University of Washington.
+ *
+ * The VIC model is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *****************************************************************************/
+
+#include <math.h>
+
+#include <vic_def.h>
+#include <vic_run.h>
+#include <vic_driver_shared.h>
+
+// FUNCTION PROTOTYPES
+double _FractionalDayFromDate(dmy_struct *date);
```

Comment:
not a fan of the change in the naming convention for functions. The only other place where we use CamelBack is in some of the snow code since that was taken without change (in the original form) from DHSVM. I'd much prefer to stick with the same naming convention for the rest of VIC. Same for the leading underscore. We don't distinguish in VIC between private and public functions and would like to stick with the same convention as before. I.e. all functions are declared in the header files, are all lower case, and don't have a leading underscore. I know there may be good reasons to do so, but staying consistent with the rest of VIC is more important.